Add Quarkus documentation (#6138)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/java/quarkus.md

software developers build efficient cloud native applications both with JVM and
Quarkus native image applications.

Quarkus uses extensions to provide optimized support for a wide range of
libraries. The
[Quarkus OpenTelemetry extension](https://quarkus.io/guides/opentelemetry)
provides:

- Out of the box instrumentation
- OpenTelemetry SDK autoconfiguration, supporting almost all system properties
  defined for the [OpenTelemetry SDK](/docs/languages/java/configuration/)
- [Vert.x](https://vertx.io/) based OTLP exporter
- The same instrumentations can be used with native image applications, which
  are not supported by the OpenTelemetry Java agent.

{{% alert-md title="Note" color="secondary" %}}

Quarkus OpenTelemetry instrumentation is maintained and supported by Quarkus.
For details, see [Quarkus community support](https://quarkus.io/support/).

{{% /alert-md %}}

Quarkus can also be instrumented with the [OpenTelemetry Java agent](../agent/)
if you are not running a native image application.

## Getting started

To enable OpenTelemetry in your Quarkus application, add the
`quarkus-opentelemetry` extension dependency to your project.

> {{< tabpane >}} {{% tab header="Maven (`pom.xml`)" lang=Maven %}}

Reviewer: ```suggestion
{{< tabpane text=true >}} {{% tab header="Maven (`pom.xml`)" lang=Maven %}}
```

---
title: Quarkus instrumentation
linkTitle: Quarkus
---

> 
> ## Quarkus OpenTelemetry instrumentation

Reviewer: The page already has a title, so let's drop this pseudo-title:

```suggestion
```

Otherwise, it looks like this:

> <img width="444" alt="image" src="https://github.com/user-attachments/assets/32f6ee75-e7fa-4313-934b-619fa79eddb5" />

[Quarkus](https://quarkus.io/) is an open source framework designed to help
software developers build efficient cloud native applications both with JVM and
Quarkus native image applications.

Quarkus uses extensions to provide optimized support for a wide range of
libraries. The
[Quarkus OpenTelemetry extension](https://quarkus.io/guides/opentelemetry)
provides:

- Out of the box instrumentation
- OpenTelemetry SDK autoconfiguration, supporting almost all system properties
  defined for the [OpenTelemetry SDK](/docs/languages/java/configuration/)
- [Vert.x](https://vertx.io/) based OTLP exporter
- The same instrumentations can be used with native image applications, which
  are not supported by the OpenTelemetry Java agent.

> > [!NOTE] Quarkus OpenTelemetry instrumentation is maintained and supported by
> > Quarkus. For details, please check the
> > [Quarkus community support page](https://quarkus.io/support/).

Reviewer: The website does not yet support GH-style alerts as you can see:

> <img width="555" alt="image" src="https://github.com/user-attachments/assets/622b673b-0c4f-41c0-ade5-1b9cd438d0f9" />

Let's use an alert here:

```suggestion
{{% alert-md title="Note" color="secondary" %}}

Quarkus OpenTelemetry instrumentation is maintained and supported by Quarkus.
For details, see [Quarkus community support](https://quarkus.io/support/).

{{% /alert-md %}}
```

Quarkus can also be instrumented with the [OpenTelemetry Java agent](../agent/)
if you are not running a native image application.

## Getting started

To enable OpenTelemetry in your Quarkus application, add the
`quarkus-opentelemetry` extension dependency to your project.

> {{< tabpane >}} {{< tab header="Maven (`pom.xml`)" lang=Maven >}}

Reviewer: Display of the tabbed content is invalid:

> <img width="444" alt="image" src="https://github.com/user-attachments/assets/ec78e735-d040-42c8-aae6-22f4bc795349" />

You need to ensure that the content gets processed as markdown, like this:

```suggestion
{{< tabpane >}} {{% tab header="Maven (`pom.xml`)" lang=Maven %}}
```

```xml
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-opentelemetry</artifactId>
</dependency>
```

{{% /tab %}} {{% tab header="Gradle (`build.gradle`)" lang=Gradle %}}

```kotlin
implementation("io.quarkus:quarkus-opentelemetry")
```

{{% /tab %}} {{< /tabpane>}}

> Only the **tracing** signal will be enabled by default. To enable **metrics**
> and **logs**, you need to add the following configuration to your
> `application.properties` file:

Reviewer: ```suggestion
Only the **tracing** signal is enabled by default. To enable **metrics** and
**logs**, add the following configuration to your `application.properties` file:
```

```properties
quarkus.otel.metrics.enabled=true
quarkus.otel.logs.enabled=true
```

> OpenTelemetry logging is available after Quarkus 3.16.0.

Reviewer: Is it after or as of?

```suggestion
OpenTelemetry logging is supported by Quarkus 3.16.0+.
```

> The remaining configurations are available in the
> [Quarkus OpenTelemetry configuration reference](https://quarkus.io/guides/opentelemetry#configuration-reference).

Reviewer: ```suggestion
For details concerning these and other configuration options, see
[OpenTelemetry Configuration Reference](https://quarkus.io/guides/opentelemetry#configuration-reference).
```

Author: We should keep `[Quarkus OpenTelemetry configuration reference]` because that page belongs to Quarkus and can cause ownership confusion.

Reviewer: There's no confusion, it links to Quarkus.io.

> ## Additional Documentation
> 
> The Quarkus documentation provides in-depth information on how to use Quarkus
> with OpenTelemetry.

Reviewer: ```suggestion
## Learn more
```

Author: To avoid ownership ambiguity we should keep: `The Quarkus documentation provides in-depth information on how to use Quarkus
with OpenTelemetry.`

> - [Using OpenTelemetry guide](https://quarkus.io/guides/opentelemetry), the
>   general reference including all
>   **[configurations](https://quarkus.io/guides/opentelemetry#configuration-reference)**.
> - [Using OpenTelemetry Tracing](https://quarkus.io/guides/opentelemetry-tracing),
>   the guide on how to **trace** your Quarkus application.
> - [Using OpenTelemetry Metrics](https://quarkus.io/guides/opentelemetry-metrics),
>   the guide about OpenTelemetry **metrics** on Quarkus applications.
> - [Using OpenTelemetry Logs](https://quarkus.io/guides/opentelemetry-logging),
>   the guide about OpenTelemetry **logs** on Quarkus applications.

Reviewer: Most link titles are descriptive enough, let's streamline this:

```suggestion
- [Using OpenTelemetry](https://quarkus.io/guides/opentelemetry), a general
  reference covering all
  [configuration](https://quarkus.io/guides/opentelemetry#configuration-reference)
  options
- Signal-specific guides for
  - [Tracing](https://quarkus.io/guides/opentelemetry-tracing)
  - [Metrics](https://quarkus.io/guides/opentelemetry-metrics)
  - [Logs](https://quarkus.io/guides/opentelemetry-logging)
```

---
title: Quarkus instrumentation
linkTitle: Quarkus
---

## Quarkus OpenTelemetry instrumentation

[Quarkus](https://quarkus.io/) is an open source framework designed to help
software developers build efficient cloud native applications both with JVM and
Quarkus native image applications.

Quarkus uses extensions to provide optimized support for a wide range of
libraries. The
[Quarkus OpenTelemetry extension](https://quarkus.io/guides/opentelemetry)
provides:

- Out of the box instrumentation
- OpenTelemetry SDK autoconfiguration, supporting almost all system properties
  defined for the
  [OpenTelemetry SDK](https://opentelemetry.io/docs/languages/java/configuration/)
- [Vert.x](https://vertx.io/) based OTLP exporter

> - The same instrumentations can be used with JVM and native image applications, where the OpenTelemetry Java agent doesn't work.

Reviewer 2: ```suggestion
- The same instrumentations can be used with native image applications, which are not supported by the OpenTelemetry Java agent.
```

> Quarkus can also be instrumented with the
> [OpenTelemetry Java agent](../agent/) but only with a JVM.

Reviewer 2: ```suggestion
Quarkus can also be instrumented with the
[OpenTelemetry Java agent](../agent/) if you are not running a native image application.
```

## Getting started

To enable OpenTelemetry in your Quarkus application, add the
`quarkus-opentelemetry` extension dependency to your project.

{{< tabpane >}} {{< tab header="Maven (`pom.xml`)" lang=Maven >}}

```xml
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-opentelemetry</artifactId>
</dependency>
```

{{% /tab %}} {{% tab header="Gradle (`build.gradle`)" lang=Gradle %}}

```kotlin
implementation("io.quarkus:quarkus-opentelemetry")
```

{{% /tab %}} {{< /tabpane>}}

Only the **tracing** signal will be enabled by default. To enable **metrics**
and **logs**, you need to add the following configuration to your
`application.properties` file:

```properties
quarkus.otel.metrics.enabled=true
quarkus.otel.logs.enabled=true
```

OpenTelemetry logging is available after Quarkus 3.16.0.

The remaining configurations are available in the
[Quarkus OpenTelemetry configuration reference](https://quarkus.io/guides/opentelemetry#configuration-reference).

{{% alert title="Note" color="info" %}}

> You can also use the [Java agent](../../agent) to instrument your Quarkus
> application in a JVM.

Reviewer 2: I don't think we need this twice, could remove either above or this note.

```suggestion
You can also use the [Java agent](../../agent) to instrument your Quarkus
application if you are not running a native image application.
```
